src/main/java/com/minecolonies/entity/EntityCitizen.java

```diff
@@ -475,11 +510,20 @@ public void onDeath(DamageSource par1DamageSource)
 
         if (colony != null)
         {
-            LanguageHandler.sendPlayersLocalizedMessage(
-              colony.getMessageEntityPlayers(),
-              "tile.blockHutTownHall.messageColonistDead",
-              citizenData.getName());
-
+            if(getColonyJob() != null && getColonyJob() instanceof JobGuard)
+            {
+                LanguageHandler.sendPlayersLocalizedMessage(
+                        colony.getMessageEntityPlayers(),
+                        "tile.blockHutTownHall.messageGuardDead",
+                        citizenData.getName());
+            }
+            else
```

Comment:
Good question, peter wanted the guard dead extra message.
But we could get the job descriptions and write it our for all the possible workers.
Might be helpful for the players.